### DISCUSSION TITLE
Inverse Backup Direction

### DISCUSSION THREAD

**Discussion by m-kreis** [oetiker/znapzend#613] [Q&A]
Hello,
I'm about to setup a backup server for our smartos host. I stumbled upon znapzend as it's now available as a precompiled package in the global zone.
Intuitively i would have expected znapzend to be able to run from the backup server. I expect a setup on which the productive system does not have any access to the backup to be more secure. In case the productive system gets corrupted (for whatever reason), there would be no way to destroy the backup as it's not accessible from the productive system.
What do you think about my concerns? Do you see any basic problems to this approach?
Best Markus

**Comment by oetiker** [comment]
I do appreciate the security concern ... it comes down to what is more 'secure' the backup location or the productive host :)

Reversing the direction would not be a fundamental problem, an additional flag in the config should do the trick. The remote backup agent would login to the server and would find the flagged backup tasks and any znapzend running centrally would ignore any flagged tasks.

BUT someone will have todo the work ... unfortunately the znapzend code has seen quite a lot of 'work' already and is in need of some re-engineering to extend it comfortably.

Do you have time and ability or a budget ?

**Comment by m-kreis** [comment]
Thanks for your message.

Yes i see. In my case i'd consider the backup location to be more secure.

Unfortunately i neither have time nor budget. It's not a really urgent matter on my side either. But it could be that i will come by this again somewhen in the future, with some time and/or budget.
